packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/converter/RectangleNodeConverterHandler.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  GQLNode,
+  GQLNodeStyle,
+  GQLRectangularNodeStyle,
+  GQLViewModifier,
+} from '../graphql/subscription/nodeFragment.types';
+import { BorderNodePositon } from '../renderer/DiagramRenderer.types';
+import { ListNodeData } from '../renderer/node/ListNode.types';
+import { RectangularNodeData } from '../renderer/node/RectangularNode.types';
+import { AlignmentMap } from './convertDiagram.types';
+import { convertLabelStyle } from './ConvertEngine';
+import { IConvertEngine, INodeConverterHandler } from './ConvertEngine.types';
+
+const defaultPosition: XYPosition = { x: 0, y: 0 };
+
+const toRectangularNode = (
+  gqlNode: GQLNode<GQLRectangularNodeStyle>,
+  gqlParentNode: GQLNode<GQLNodeStyle> | null,
+  isBorderNode: boolean
+): Node<RectangularNodeData> => {
+  const {
+    targetObjectId,
+    targetObjectLabel,
+    targetObjectKind,
+    descriptionId,
+    id,
+    insideLabel,
+    state,
+    style,
+    labelEditable,
+  } = gqlNode;
+
+  const data: RectangularNodeData = {
+    targetObjectId,
+    targetObjectLabel,
+    targetObjectKind,
+    descriptionId,
+    style: {
+      display: 'flex',
+      backgroundColor: style.color,
+      borderColor: style.borderColor,
+      borderRadius: style.borderRadius,
+      borderWidth: style.borderSize,
+      borderStyle: style.borderStyle,
+    },
+    label: undefined,
+    faded: state === GQLViewModifier.Faded,
+    isBorderNode: isBorderNode,
+    borderNodePosition: isBorderNode ? BorderNodePositon.EAST : null,
+    labelEditable,
+  };
+
+  if (insideLabel) {
+    const labelStyle = insideLabel.style;
+    data.label = {
+      id: insideLabel.id,
+      text: insideLabel.text,
+      isHeader: insideLabel.isHeader,
+      style: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        justifyContent: 'center',
+        padding: '8px 16px',
+        textAlign: 'center',
+        ...convertLabelStyle(labelStyle),
+      },
+      iconURL: labelStyle.iconURL,
+    };
+
+    const alignement = AlignmentMap[insideLabel.insideLabelLocation];
+    if (alignement.isPrimaryVerticalAlignment) {
+      if (alignement.primaryAlignment === 'TOP') {
+        if (data.label.isHeader) {
+          data.label.style.borderBottom = `${style.borderSize}px ${style.borderStyle} ${style.borderColor}`;
+        }
+        data.style = { ...data.style, display: 'flex', flexDirection: 'column', justifyContent: 'flex-start' };
+      }
+      if (alignement.secondaryAlignment === 'CENTER') {
+        data.style = { ...data.style, alignItems: 'stretch' };
+        data.label.style = { ...data.label.style, justifyContent: 'center' };
+      }
+    }
+  }
+
+  const node: Node<RectangularNodeData> = {
+    id,
+    type: 'rectangularNode',
+    data,
+    position: defaultPosition,
+    hidden: state === GQLViewModifier.Hidden,
+  };
+
+  if (gqlParentNode) {
+    node.parentNode = gqlParentNode.id;
+  }
+
+  return node;
+};
+
+const toListNode = (
```

Comment:
Why keep this in `RectangleNodeConverter`? Having 1 file / 1 ConverterHandler per type would be more regular.

Reply:
For these converter, we only take into account the style name and for list node, it's actualy a `RectangularNodeStyle` with a specific childrenLayoutStrategy.
We could move this check in the `canHandle` to split the rectangular in two converters, it's done in the seperate commit.